src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -1,15 +1,1553 @@
+
+    if (SQ(playerX - thisX) + SQ(playerZ - thisZ) < SQ(80.0f)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void func_80AB36DC(EnNb* this, GlobalContext* globalCtx) {
+    u16 moveTime = this->movementTimer;
+
+    if ((((u16)((u16)(kREG(17) + 25) - 4))) > moveTime) {
+        s16 invScale = 4 - moveTime;
+        if (invScale > 0) {
+            Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->pathYaw, invScale, 6200, 100);
+        }
+    } else {
+        s16 invScale = (u16)(kREG(17) + 25) - moveTime;
+        if (invScale > 0) {
+            Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.initPosRot.rot.y, invScale, 6200, 100);
+        }
+    }
+}
+
+void EnNb_CheckNoticed(EnNb* this, GlobalContext* globalCtx) {
+    if (EnNb_GetNoticedStatus(this, globalCtx)) {
+        EnNb_SetCurrentAnim(this, &D_06004E60, 2, -8.0f, 0);
+        this->action = NB_NOTICE_PLAYER;
+        EnNb_SetNoticeSFX(this);
+    }
+}
+
+void EnNb_SetupIdleCrawlspace(EnNb* this, s32 animFinished) {
+    if (animFinished) {
+        EnNb_SetCurrentAnim(this, &D_06004BB4, 0, -8.0f, 0);
+        this->headTurnFlag = 1;
+        this->actor.flags |= 9;
+        this->action = NB_IDLE_CRAWLSPACE;
+    }
+}
+
+void func_80AB3838(EnNb* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx)) {
+        this->action = NB_IN_DIALOG;
+    } else {
+        this->actor.flags |= 9;
+        
+        if (!(gSaveContext.infTable[22] & 0x1000)) {
+            this->actor.textId = 0x601D;
+        } else {
+            this->actor.textId = 0x6024;
+        }
+
+        func_8002F2F4(&this->actor, globalCtx);
+    }
+}
+
+void EnNb_SetupPathMovement(EnNb* this, GlobalContext* globalCtx) {
+    EnNb_SetCurrentAnim(this, &D_06004E60, 2, -8.0f, 0);
+    gSaveContext.eventChkInf[9] |= 0x10;
+    this->action = NB_IN_PATH;
+    this->actor.flags &= ~9;
+}
+
+void EnNb_SetTextIdAsChild(EnNb* this, GlobalContext* globalCtx) {
+    s32 pad;
+    u8 choiceIndex;
+    s32 pad1;
+    u16 textId;
+
+    textId = this->actor.textId;
+
+    if (func_8010BDBC(&globalCtx->msgCtx) == 2) {
+        if (textId == 0x6025) {
+            EnNb_SetupPathMovement(this, globalCtx);
+        } else {
+            if (textId == 0x6027) {
+                gSaveContext.infTable[22] |= 0x1000;
+            }
+            this->action = NB_IDLE_CRAWLSPACE;
+        }
+        this->actor.flags &= ~9;
+    } else if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && (func_80106BC8(globalCtx) != 0)) {
+        choiceIndex = globalCtx->msgCtx.choiceIndex;
+        
+        if (textId == 0x601D) {
+            switch (choiceIndex) {
+                case 0:
+                    this->actor.textId = 0x601E;
+                    break;
+                case 1:
+                    this->actor.textId = 0x601F;
+                    break;
+                default:
+                    this->actor.textId = 0x6020;
```

Comment:
```suggestion
                    this->actor.textId = 0x6020;
                    break;
```

Reply:
doesnt match, oddly

Reply:
adding a break gets rid of an uncondional branch and makes all other branches go to a different target, kinda weird

Reply:
Very strange indeed, nvm then